lib/charms/kubernetes_charm_libraries/v0/kubernetes_volumes_patch.py

```diff
@@ -0,0 +1,530 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Charm Library used to leverage the Volumes Kubernetes in charms.
+
+- On config-changed, it will:
+  - Patch the StatefulSet with the required volumes
+  - Patch the Pod with the required volume mounts and resources limits when necessary
+
+## Usage
+
+```python
+
+from charms.kubernetes_charm_libraries.v0.volumes import (
+    KubernetesMultusCharmLib,
+    RequestedVolume
+)
+
```

Comment:
We can add sample custom event definitions and how charm emits the event when required.